### PR TITLE
a-o-i: Move oo_config objects out

### PR DESCRIPTION
**Pull request by smunilla**
Move the model objects used in oo_config out to their own models
file for readablity and future expansion.

### CODE REVIEW

**Comment by smunilla** [comment]
aos-ci-test

**Comment by smunilla** [comment]
@tbielawa Does this conflict with any of the work you've been doing?

File: utils/test/oo_config_tests.py [hunk 1]
```diff
@@ -3,12 +3,14 @@
...
 import os
-import unittest
-import tempfile
 import shutil
+import tempfile
+import unittest
+
 import yaml
+from ooinstall.models import InvalidHostError
+from ooinstall.oo_config import Host, OOConfig
```
Comment (tbielawa) [inline utils/test/oo_config_tests.py]: I'm not sure why this doesn't raise errors when I run CI on it. `ooinstall.oo_config` shouldn't have `Host` in it anymore. I'd think this should be more like this:

> from ooinstall.models import InvalidHostError, Host
> from ooinstall.oo_config import OOConfig

File: utils/src/ooinstall/cli_installer.py [hunk 2]
```diff
@@ -10,7 +10,7 @@
 from pkg_resources import parse_version
 from ooinstall import openshift_ansible
 from ooinstall.oo_config import OOConfig
-from ooinstall.oo_config import OOConfigInvalidHostError
+from ooinstall.models import InvalidHostError
 from ooinstall.oo_config import Host, Role
```
Comment (tbielawa) [inline utils/src/ooinstall/cli_installer.py]: Still not sure how this isn't erroring. Shouldn't `Host` and `Role` be imported from `ooinstall.models` now?

Comment (smunilla) [inline utils/src/ooinstall/cli_installer.py]: Very strange, I'll have to assume some kind of chained imports are in place here, but I'll fix it regardless.

**Comment by smunilla** [comment]
aos-ci-test

**Comment by smunilla** [comment]
@tbielawa Updated to avoid the previous insanity. Also a lot of pylint fixes since I was already messing around with stuff.

**Comment by tbielawa** [comment]
I'll drop in a formal review once the conflicts are fixed

**Comment by smunilla** [comment]
Conflicts fixed!

**Comment by smunilla** [comment]
aos-ci-test

**Comment by tbielawa** [comment]
@smunilla got some pylint to fix on this one here

**Comment by smunilla** [comment]
aos-ci-test

**Comment by mtnbikenc** [comment]
Closing this PR as it is stale and needs to be rebased.   Please reopen the PR if you feel this patch is still applicable.  Thanks!

**Pull request closed without merging**